pkg/operator/controller/ttl_report.go

```diff
@@ -63,12 +66,8 @@ func (r *TTLReportReconciler) reconcileReport() reconcile.Func {
 		if err != nil {
 			return ctrl.Result{}, fmt.Errorf("failed parsing %v with value %v %w", v1alpha1.TTLReportAnnotation, ttlReportAnnotationStr, err)
 		}
-		creationTime := report.Report.UpdateTimestamp
-		ttlExpired, durationToTTLExpiration, err := ttlIsExpired(reportTTLTime, creationTime.Time)
-		if err != nil {
-			return ctrl.Result{}, err
-		}
-		if ttlExpired {
+		durationToTTLExpiration := utils.NextIntervalDuration(reportTTLTime, report.Report.UpdateTimestamp.Time, r.Clock)
```

Comment:
TBH, this code was much more readable with `ttlIsEExpired` method that accepted 2 args. Do you think we can add `utils.IsTTLExpired(clock ext.Clock, creationTime time.Time, ttl time.Duration) (bool, time.Duration)` to the utils package?

Reply:
sure, I prefer to add it to ttl_report controller as dateUtil should be agnostic to logic terms (a.k.a: ttl)

Reply:
sure can do that, initially didn't want date util to include logic terms (ttl)